include/calibrationLib.py

```diff
@@ -251,4 +258,51 @@ def query_betas(connection, studyId, filterZero=True, filename="data/calibration
         csvfile.write(header)
         for row in rows:
             line = ','.join(str(row[ndx]) for ndx in range(0, len(row)))
-            csvfile.write("{}\n".format(line))
\ No newline at end of file
+            csvfile.write("{}\n".format(line))
+
+def goodness_of_variance_fit(array, classes):
+    # get the break points
+    classes = jenkspy.jenks_breaks(array, classes)
+
+    # do the actual classification
+    classified = np.array([classify(i, classes) for i in array])
+
+    # max value of zones
+    maxz = max(classified)
+
+    # nested list of zone indices
+    zone_indices = [[idx for idx, val in enumerate(classified) if zone + 1 == val] for zone in range(maxz)]
+
+    # sum of squared deviations from array mean
+    sdam = np.sum((array - array.mean()) ** 2)
+
+    # sorted polygon stats
+    array_sort = [np.array([array[index] for index in zone]) for zone in zone_indices]
+
+    # sum of squared deviations of class means
+    sdcm = sum([np.sum((classified - classified.mean()) ** 2) for classified in array_sort])
+
+    # goodness of variance fit
+    gvf = (sdam - sdcm) / sdam
+
+    return gvf, classes
+
+def classify(value, breaks):
+    for i in range(1, len(breaks)):
+        if value < breaks[i]:
+            return i
+    return len(breaks) - 1
+
+# getting data ready for binning
+def data_bin(filename):
+
+    # data should be 1-dimensional array, python list or iterable
+
+    with open(filename) as ascfile:
+        myArray = np.loadtxt(ascfile, skiprows=6)
+
+    myArray[myArray == -9999] = np.nan
+    myArray = myArray[~np.isnan(myArray)]
```

Comment:
Remember, you are working with ASC data which is a defined file type, which is why using a loader for the type is the preferred option since the header defines what the `nodata` value is. While `-9999` is being used as the `nodata` value here, it's entirely possible that the raster files for the next project will use a different value.

Reply:
That's what I was thinking why we have `-9999` as `no data` value. Can you tell if `nodata` value for other files is always less than `0`.

Reply:
Also let me know what is the criteria for determining the value of `nodata`. Like for `populationBins.asc`, how is the value -9999 determined? Why is it only -9999 and not any other value for ex., -22, -30, -5000 ?

Reply:
@PakhiAgarwal There's no real way to know what it will be, but that's why the value is supplied in the header:

```python
[header, data] = load_asc(filename)
nodata = header['nodata']
```

Reply:
Okay, so it means `nodata` value can either be positive or negative. 
If it is positive then what happens if it clashes with other values, for ex., if `nodata` is say `20`, then can we have `20` as other actual `real value`?

Reply:
@PakhiAgarwal If the `nodata` value clashes with a real data value then it means the file is not correct and there are likely other problems that we would be worried about.